env0/data_aws_cost_credentials.go

```diff
@@ -0,0 +1,92 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataAwsCostCredentials() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataAwsCostCredentialsRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "the name of the credential",
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+			"id": {
+				Type:         schema.TypeString,
+				Description:  "the id of the credential",
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+		},
+	}
+}
+
+func dataAwsCostCredentialsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var err diag.Diagnostics
+	var credentials client.Credentials
+
+	id, ok := d.GetOk("id")
+	if ok {
+		credentials, err = getAwsCostCredentialsById(id.(string), meta)
+		if err != nil {
+			return err
+		}
+	} else {
+		name, ok := d.GetOk("name")
+		if !ok {
+			return diag.Errorf("Either 'name' or 'id' must be specified")
+		}
+		credentials, err = getAwsCostCredentialsByName(name.(string), meta)
+		if err != nil {
+			return err
+		}
+	}
+
+	d.SetId(credentials.Id)
+	d.Set("name", credentials.Name)
```

Comment:
Please use the new common `writeResourceData` instead. Have a look at existing usages of it to get the idea.

Note: this comment is relevant to all other resources

Reply:
Good point ill do it, just a little note, the function you mention based on the name of the field in schema against the field name in the struct , in aws creds we have 'arn' in schema but the struct field is '**RoleArn** so in some places like that we cant use those functions, or we can just change the name of the field in struct/schema

Reply:
4eec18f9488a26f0acf5fe31b46a44b23cc303cd

Reply:
So how did you solve that arn -> RoleArn tranform?

Reply:
In this specific code there is no `arn` attribute so I do used the function you suggested , but in resource(no data resource) for example I cant use it, another problem in this function is when the value of the filed in the struct is struct itself like `value` in `awscredentialscreate` but for this I have a solution, anyway its not enough because the problem we talked above, unless we change the schema field name to `role_arn`

Reply:
Ok understood. 
Regarding your last sentence - I do think we should change it to `role_arn` so:
1. api and schema are aligned 
2. We can use this util